crates/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -129,12 +129,26 @@ impl CrateDefMap {
     pub fn get_all_test_functions<'a>(
         &'a self,
         interner: &'a NodeInterner,
-    ) -> impl Iterator<Item = FuncId> + 'a {
+    ) -> impl Iterator<Item = TestFunc> + 'a {
         self.modules.iter().flat_map(|(_, module)| {
             module
                 .value_definitions()
                 .filter_map(|id| id.as_function())
-                .filter(|id| interner.function_meta(id).attributes == Some(Attribute::Test))
+                .filter(|id| {
+                    interner.function_meta(id).attributes
+                        == Some(Attribute::Test { expect_failure: true })
+                        || interner.function_meta(id).attributes
+                            == Some(Attribute::Test { expect_failure: true })
+                })
```

Comment:
```suggestion
                .filter(|id| {
                    matches!(interner.function_meta(id).attributes, Some(Attribute::Test { .. }))
                })
```
`matches` should clean this up